macros/sql/get_column_from_sql.sql

```diff
@@ -0,0 +1,3 @@
+{% macro get_column_from_sql(sql, column_name) %}
+    {{ return(dbt_utils.fetch(sql)[column_name]) }}
```

Comment:
one of the big challenges associated with dbt's current compilation scheme is that macros are called twice, once in parsing (when `execute` is false) and then again when dbt is _really_ running the macro.

The implementation of `get_column_sql` shown here will work at _runtime_ but it will return `none` at parse-time, which probably won't be expected by the caller. If we return a `table` in the `fetch` macro, then we can pluck out the specified column in `get_column_from_sql`. If the table returned by fetch is `none` (ie. we're in parsing), then we can just return an empty list here! That should simplify the implementation logic for callers of this macro

Reply:
I still don't get this comment...